firesteel/src/main/scala/org/apache/spark/shuffle/shm/ShmShuffleManager.scala

```diff
@@ -110,9 +110,7 @@ private[spark] class ShmShuffleManager (conf: SparkConf) extends ShuffleManager
   override def unregisterShuffle(shuffleId: Int): Boolean = {
     if (_shuffleStoreManager != null) {
       logInfo("shm-shuffle store manager cleanup map stores with shuffle id: " + shuffleId)
-      //NOTE: 2/23/2016, to comment out the cleanup invocation temporarily, as for large-scale
-      //BP application testing on DragonHawk, it does not work.
-      //shuffleStoreManager.cleanup(shuffleId)
+      _shuffleStoreManager.cleanup(shuffleId)
```

Comment:
the above comment says cleanup did not work, did you try this with larger applications. If so, please share the logs.

Reply:
Thank you for your hard work.
I am not really sure the definition of larger applications(and also DragonHawk) but I have already run the fixed program in 4 processors(not cores) NUMA machine. 
What logs would you like to be shared.

Reply:
The below code is from SortShuffleManager of Sparkle 2.4, here  SortShuffleManager unregister does cleanup of all the maps,  did you explore this option?

/** Remove a shuffle's metadata from the ShuffleManager. */
  override def unregisterShuffle(shuffleId: Int): Boolean = {
    Option(numMapsForShuffle.remove(shuffleId)).foreach { numMaps =>
      (0 until numMaps).foreach { mapId =>
        shuffleBlockResolver.removeDataByMap(shuffleId, mapId)
      }
    }
    true
  }